### PR TITLE
Move implementation inside `runTransactionCmd` to toplevel definitions.

### PR DESCRIPTION
**Pull request by MarcFontaine**

### CODE REVIEW

**Comment by MarcFontaine** [comment]
https://github.com/input-output-hk/cardano-node/issues/4672

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs [hunk 1]
```diff
@@ -447,6 +302,215 @@ runTransactionCmd cmd =
 -- Building transactions
 --
 
+runTxBuildCmd
+  :: IsCardanoEra era
+  => CardanoEra era
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs:307]: LGTM, however lets use `AnyCardanoEra` here and add some haddock annotations to the parameters where it's not obvious what they are e.g `Maybe Word`.  The haddocks don't have to be overly descriptive, short succinct ones will do.

**Review by Jimbo4350 — CHANGES_REQUESTED** [review]
LGTM but one comment

**Review by Jimbo4350 — APPROVED** [review]
LGTM!

**Comment by MarcFontaine** [comment]
https://github.com/input-output-hk/cardano-node/issues/4672

**Pull request merged**